aten/src/ATen/native/transformers/cuda/sdp_utils.cpp

```diff
@@ -176,11 +177,42 @@ bool check_sm_version(cudaDeviceProp * dprops) {
   return is_gte_lower_bound && is_lte_upper_bound;
 }
 
+#if USE_ROCM
+c10::once_flag gcn_arch_override_flag;
+const char* over_arch = nullptr;
+
+void init_gcn_arch_override() {
+  over_arch = std::getenv("PYTORCH_DEBUG_FLASH_ATTENTION_GCN_ARCH_OVERRIDE");
+  if (over_arch) {
+      TORCH_WARN("SDPA functions only loads value from PYTORCH_DEBUG_FLASH_ATTENTION_GCN_ARCH_OVERRIDE once. "
+                 "Later changes to this environment variable with os.environ "
+                 "(or other methods) will not affect SDPA function's behavior.");
+  }
+}
+#endif
+
 bool check_flash_attention_hardware_support(sdp_params const& params, bool debug) {
   // Check that the gpu is capable of running flash attention
   using sm80 = SMVersion<8, 0>;
   using sm90 = SMVersion<9, 0>;
   auto dprops = at::cuda::getCurrentDeviceProperties();
+#if USE_ROCM
+  constexpr std::string_view mi200 = "gfx90a:sramecc+:xnack-";
+  const char* real_arch = dprops->gcnArchName;
+  c10::call_once(gcn_arch_override_flag, init_gcn_arch_override);
```

Comment:
Considering that `over_arch` is only used inside  `check_flash_attention_hardware_support` it feels like it would be much more convenient (and aligned with C++11 recommendation) to write something like the follwing
```suggestion
  static const char *over_arch = [] {
   auto rc  = std::getenv("PYTORCH_DEBUG_FLASH_ATTENTION_GCN_ARCH_OVERRIDE");
   if (rc) {
       TORCH_WARN("SDPA functions only loads value from 
                 PYTORCH_DEBUG_FLASH_ATTENTION_GCN_ARCH_OVERRIDE once. "
                 "Later changes to this environment variable with os.environ "
                 "(or other methods) will not affect SDPA function's behavior.");
    }
    return rc;
  }();
```